packages/pkp-walletconnect/src/lib/pkp-walletconnect.ts

```diff
@@ -0,0 +1,638 @@
+   *
+   * @returns {boolean} - True if chain is supported, false otherwise
+   */
+  public checkIfChainIsSupported(chain: string): boolean {
+    const parsedChain = parseChainId(chain);
+    const chainId = Number.parseInt(parsedChain.reference, 10);
+    // TODO: Update this once we support more JSON RPC handlers
+    if (parsedChain.namespace === 'eip155') {
+      for (const key in LIT_CHAINS) {
+        const chain = LIT_CHAINS[key];
+        if (chain.chainId === chainId) {
+          return true;
+        }
+      }
+      return false;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Find PKPClient by request event params
+   *
+   * @param {any} params - Request event params
+   */
+  public async findPKPClientByRequestParams(
+    params: any
+  ): Promise<PKPClient | null> {
+    const paramsString = JSON.stringify(params);
+
+    // Loop through all wallets and find the one that has an address that can be found within the request params
+    for (const pkpClient of this.pkpClients) {
+      // TODO: Update this once we support more JSON RPC handlers
+      const ethWallet = pkpClient.getEthWallet();
+      const ethAddress = await ethWallet.getAddress();
+      if (paramsString.toLowerCase().includes(ethAddress.toLowerCase())) {
+        return pkpClient;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Add a PKPClient to list of PKPClients if not already added
+   *
+   * @param {PKPClient} pkpClient - The PKPClient instance
+   */
+  public addPKPClient(pkpClient: PKPClient): void {
+    const existingClient = this.pkpClients.find(
+      (client) => client.pkpPubKey === pkpClient.pkpPubKey
+    );
+    if (!existingClient) {
+      this.pkpClients.push(pkpClient);
+    }
+  }
+
+  /**
+   * Get current list of PKPClients
+   *
+   * @returns {PKPClient[]} - List of PKPClients
+   */
+  public getPKPClients(): PKPClient[] {
+    return this.pkpClients;
+  }
+
+  // ----------------- Private methods -----------------
+
+  /**
+   * Checks if the given WalletConnect client is initialized and returns it. If it's not initialized, throws an error.
+   *
+   * @private
+   * @param {IWeb3Wallet | undefined} client - The WalletConnect client instance to check for initialization
+   * @returns {IWeb3Wallet} - The initialized WalletConnect client instance
+   * @throws {Error} - If the WalletConnect client instance is not initialized
+   */
+  private _isWalletConnectInitialized(
+    client: IWeb3Wallet | undefined
+  ): IWeb3Wallet {
+    if (!client) {
+      this._log('WalletConnect client has not yet been initialized.');
+      return this._throwError(
+        'WalletConnect client has not yet been initialized. Please call initWalletConnect().'
+      );
+    }
+    return client;
+  }
+
+  /**
+   * Logs the provided arguments to the console if the `debug` property is set to true.
+   *
+   * @private
+   * @param {...any[]} args - The arguments to log to the console.
+   */
+  private _log(...args: any[]): void {
```

Comment:
do we need similar methods for other log levels? Should we use other log levels?